src/librustc_mir/interpret/operand.rs

```diff
@@ -182,6 +173,20 @@ impl<'tcx> From<ValTy<'tcx>> for OpTy<'tcx> {
     }
 }
 
+// Validation needs to hash OpTy, but we cannot hash Layout -- so we just hash the type
+impl<'tcx> Hash for OpTy<'tcx> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.op.hash(state);
+        self.layout.ty.hash(state);
+    }
+}
+impl<'tcx> PartialEq for OpTy<'tcx> {
+    fn eq(&self, other: &Self) -> bool {
+        self.op == other.op && self.layout.ty == other.layout.ty
```

Comment:
This is a similar problem as with #53424. Maybe we should just derive `PartialEq, Eq, Hash` on `Layout` and `TyLayout`?!

Reply:
It is the same problem in fact.

Why have they not been derived in the first place?

Reply:
I think they're more expensive to derive if they include the actual layout.

Reply:
(I got a notification for you writing a comment here but now cannot see the comment, @eddyb... wtf?)

FWIW, in the mean time https://github.com/rust-lang/rust/pull/54936 has landed.

Reply:
I deleted the comment after seeing that PR.